lib/active_model_serializers/adapter/json_api.rb

```diff
@@ -498,8 +498,8 @@ def links_for(serializer)
       #   end
       # prs:
       #   https://github.com/rails-api/active_model_serializers/pull/1041
-      def pagination_links_for(serializer, options)
-        PaginationLinks.new(serializer.object, options[:serialization_context]).serializable_hash(options)
+      def pagination_links_for(serializer)
+        PaginationLinks.new(serializer.object, instance_options[:serialization_context]).as_json(instance_options)
```

Comment:
(which is what was wrong with the tests that were trying to pass in AMS options to `serializable_hash` in the tests changed in this diff).

Also, not having controller tests made this misunderstanding a bit easier to happen